packages/chain-adapters/src/solana/SolanaChainAdapter.ts

```diff
@@ -342,7 +359,38 @@ export class ChainAdapter implements IChainAdapter<KnownChainIds.SolanaMainnet>
     this.providers.ws.close('txs')
   }
 
-  protected async parseTx(tx: unchained.solana.Tx, pubkey: string): Promise<Transaction> {
+  private buildEstimationSerializedTx(input: GetFeeDataInput<KnownChainIds.SolanaMainnet>): string {
+    const instructions = input.chainSpecific.instructions ?? []
+
+    const value = Number(input.value)
+    if (!isNaN(value) && value > 0 && input.to) {
+      instructions.push(
+        SystemProgram.transfer({
+          fromPubkey: new PublicKey(input.chainSpecific.from),
+          toPubkey: new PublicKey(input.to),
+          lamports: value,
+        }),
+      )
+    }
+
+    // max compute unit limit instruction for fee estimation
+    instructions.push(ComputeBudgetProgram.setComputeUnitLimit({ units: 1400000 }))
+    // placeholder compute unit price instruction for fee estimation
+    instructions.push(ComputeBudgetProgram.setComputeUnitPrice({ microLamports: 0 }))
```

Comment:
correct, we just want the instruction when estimating so we estimate the correct number of compute units. It may be set to 0 in the final transaction, but with no priority fee set, there won't be an extra charge for the extra compute units